### PR TITLE
[stage 1] Token replacement + autocomplete

### PR DESCRIPTION
**Pull request by lizziegooding**
### Context
Right now, we replace tokens in the user’s query, then autocomplete on the token-replaced phrases. This leads to unexpected behavior when the replaced token is at the end of the user’s query where we consider possible matches where the things we’ve replaced is the *beginning* of another word. For example: :

| User’s query      | Token replacement | Autocomplete match | Result returned to the user |
| ----------------- | ----------------- | ------------------ | --------------------------- |
| "sunset district” | "sunset”, “dt”    | "dt" → "dtown”     | "dtown party bus”           |
| "the white house" | "the", "white", "ho" | "ho" → "horse" | "the white horse" |
| "saint street" | "st", "st" | "st" → "stephen" |  "saint stephen" |
| "court house" | "ct", "ho" | "ho" → "hotel" | "court hotel" |
| "the house" | "the", "ho" | "ho" → "hound" | "The Hound Gentlemen's Clothiers" |
| "chase" | "ch" | "ch" → "chicago" | Chicago, IL |
|  "delaware county" | "delaware", "co" | "co" → "college" | "university of delaware" |

### Summary of Changes
- [x] adds acceptance test with `district` test case

### Next Steps
- [ ]  Track whether the last phrase in the query was token-replaced in carmen’s replaceToken function

cc @miccolis @aarthykc @apendleton @boblannon

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -5,14 +5,12 @@
 "@babel/code-frame@^7.0.0":
   version "7.0.0"
   resolved "https://registry.yarnpkg.com/@babel/code-frame/-/code-frame-7.0.0.tgz#06e2ab19bdb535385559aabb5ba59729482800f8"
-  integrity sha512-OfC2uemaknXr87bdLUkWog7nYuliM9Ij5HUcajsVcMCpQrcLmtxRbVFTIqmcSkSeYRBFBRxs2FiUqFJDLdiebA==
```
Comment (miccolis) [inline yarn.lock]: Seems like we shouldn't be removing all these `integrity` things. Any idea what's up here?

File: lib/text-processing/token.js [hunk 2]
```diff
@@ -103,17 +105,39 @@ module.exports.createReplacer = function(tokens, inverseOpts) {
...
+                false;
 
             if (entry.named) {
-                entry.from = new XRegExp('(?<tokenBeginning>' + WORD_BOUNDARY + '|^)' + from.replace('.', '\\.') + '(?<tokenEnding>' + WORD_BOUNDARY + '|$)', 'gi');
+                entry.from = new XRegExp(
+                    '(?<tokenBeginning>' + WORD_BOUNDARY + '|^)' +
+                    from.replace('.', '\\.') +
+                    '(?<tokenEnding>' + WORD_BOUNDARY + '|$)',
+                    'gi');
+                entry.fromLastWord = complex ? false :
```
Comment (miccolis) [inline lib/text-processing/token.js]: Do we have it documented right now that this can be either a boolean or a Regexp? It definitely confused me.

Comment (lizziegooding) [reply lib/text-processing/token.js]: Good call, I'll add to the JS docs

Comment (lizziegooding) [reply lib/text-processing/token.js]: Added in https://github.com/mapbox/carmen/pull/795/files#diff-d7e5e85dadadd9e3925abd662f2ee245R16

File: test/unit/text-processing/token.createReplacer.test.js [hunk 3]
```diff
@@ -0,0 +1,162 @@
+'use strict';
```
Comment (lizziegooding) [inline test/unit/text-processing/token.createReplacer.test.js]: I `git mv`'d this locally from ` test/unit/text-processing/token.test.js` -> `test/unit/text-processing/token.createReplacer.test.js` -- not sure why this worked for replace.test.js but not this file

File: test/unit/text-processing/token.replaceToken.test.js [hunk 4]
```diff
@@ -317,91 +319,75 @@ test('custom reverse replacement', (q) => {
         'East first street'
     ]);
 
-    q.end();
+    t.end();
 });
 
-test('replacer', (q) => {
```
Comment (lizziegooding) [inline test/unit/text-processing/token.replaceToken.test.js]: I moved this test to https://github.com/mapbox/carmen/pull/795/files#diff-94a779fff94b500c61fa5d097db04964R13 as it tested the same functionality. `Monsieur -> M` is a simple token replacement and should behave exactly the same as these `Street -> St` and `Road -> Rd` examples.

**Pull request merged**